python/setup.py

```diff
@@ -82,31 +93,42 @@ def run(self):
                 os.path.join("ray/pyarrow_files/pyarrow", filename))
         ]
 
-        files_to_include = ray_files + pyarrow_files
+        other_files_to_include = pyarrow_files
 
         # Copy over the autogenerated flatbuffer Python bindings.
         for directory in generated_python_directories:
             for filename in os.listdir(directory):
                 if filename[-3:] == ".py":
-                    files_to_include.append(os.path.join(directory, filename))
+                    other_files_to_include.append(
+                        os.path.join(directory, filename))
 
-        for filename in files_to_include:
-            self.move_file(filename)
+        for filename in other_files_to_include:
+            self.move_file(filename, os.path.join(self.build_lib, filename))
 
-        # Try to copy over the optional files.
+        # Try to copy over the generated optional files.
+        for filename in optional_ray_files_generated:
+            try:
+                source = "../build/" + filename
+                destination = "ray/core/" + filename
+                self.move_file(source, os.path.join(self.build_lib,
+                                                    destination))
+                self.move_file(source, destination)
```

Comment:
Why are there two calls to `move_file` here? I would expect the handling for `other_files_to_include` and `optional_ray_files_generated` to be the same.

Reply:
I move the executables to two places before, now I modify the CMakeLists.txt file to enable the bins and libs can automatically installed to the original place . so now only one `move_file` needed.